api-catalog-services/src/test/java/org/zowe/apiml/apicatalog/standalone/StandaloneLoaderServiceTest.java

```diff
@@ -0,0 +1,172 @@
+        private String testData;
+
+        @BeforeEach
+        void init() throws IOException {
+            GatewayClient gatewayClient = new GatewayClient(null);
+            TransformService transformService = new TransformService(gatewayClient);
+            cachedServicesService = new CachedServicesService();
+            CachedProductFamilyService cachedProductFamilyService =
+                new CachedProductFamilyService(cachedServicesService, transformService, null);
+            InstanceInitializeService instanceInitializeService = new InstanceInitializeService(cachedProductFamilyService, cachedServicesService, null, null);
+            StandaloneAPIDocRetrievalService standaloneAPIDocRetrievalService = new StandaloneAPIDocRetrievalService();
+            cachedApiDocService = new CachedApiDocService(standaloneAPIDocRetrievalService, null);
+            standaloneLoaderService =
+                new StandaloneLoaderService(new ObjectMapper(), instanceInitializeService, cachedApiDocService, standaloneAPIDocRetrievalService);
+
+            testData = setTestData("standalone/services");
+            cachedApiDocService.resetCache();
+        }
+
+        @Test
+        void thenContainerCacheIsPopulated() {
+            assertNull(cachedServicesService.getAllCachedServices());
+
+            standaloneLoaderService.initializeCache();
+
+            assertEquals(3, cachedServicesService.getAllCachedServices().size());
+            assertEquals("SERVICE1-INSTANCE1", cachedServicesService.getAllCachedServices().getRegisteredApplications("service1-instance1").getName());
+            assertEquals("SERVICE1-INSTANCE2", cachedServicesService.getAllCachedServices().getRegisteredApplications("service1-instance2").getName());
+            assertEquals("SERVICE2", cachedServicesService.getAllCachedServices().getRegisteredApplications("service2").getName());
+        }
+
+        @Test
+        void thenApiDocCacheIsPopulated() throws IOException {
+            String expectedService1Instance1apiDoc = readTestData("/apiDocs/service1-instance1_zowe v1.0.0_default.json");
+            String expectedService1Instance2apiDoc = readTestData("/apiDocs/service1-instance2_zowe v1.0.0_default.json");
+            String expectedService2apiDoc1 = readTestData("/apiDocs/service2_zowe v1.0.0_default.json");
+            String expectedService2apiDoc2 = readTestData("/apiDocs/service2_zowe v2.0.0.json");
+
+            assertThrows(ApiDocNotFoundException.class, () -> cachedApiDocService.getDefaultApiDocForService("service1-instance1"));
+            assertThrows(ApiDocNotFoundException.class, () -> cachedApiDocService.getDefaultApiDocForService("service1-instance2"));
+            assertThrows(ApiDocNotFoundException.class, () -> cachedApiDocService.getDefaultApiDocForService("service2"));
+
+            assertThrows(ApiDocNotFoundException.class, () -> cachedApiDocService.getApiDocForService("service1-instance1", "zowe v1.0.0"));
```

Comment:
Is the Sonar reclaim of using the same string multiple times (more than 3 times)?

Reply:
Fortunately no :-) I was considering it but I find the code more readable if the string is directly in assert.